### DISCUSSION TITLE
Pydantic refuse to validate my schema

### DISCUSSION THREAD

**Discussion by goldyfruit** [fastapi/fastapi#6750] [Questions] [answered]
## Description

Hi,

I'm pretty new with Pydantic validation but it makes me think a lot to Golang structures.
I'm trying to `POST` a pretty simple JSON payload but Pydantic keeps refusing the validation when using `curl`.

```json
{
  "target": "https://google.fr",
  "link": "BHDH",
  "extras": {
    "code": "ABCDE"
  }
}
```

Here is my `schemas.py`.
```python
from typing import Optional, Any, Dict
from pydantic import BaseModel, HttpUrl, Json

class Link(BaseModel):
    link: str
    target: httpUrl
    extras: Json[Dict[str, Any]]

print(Link(link='BHDH', target='https://google.com, extras='{"code": "ABCDE"}'))
```

Here is the `curl` command used to `POST` the JSON payload.
```
$ curl -X POST http://localhost:8000 -d '{"target": "https://google.fr", "link": "BHDH", "extras": {"code": "ABCDE"}}'
```
And this is the error I got:
```
{"detail":[{"loc":["body","extras"],"msg":"JSON object must be str, bytes or bytearray","type":"type_error.json"}]}
```

Thanks for your help :)

## Environment

    OS: Linux
    FastAPI Version: 0.63.0
    Python-version: 3.9.2

**Comment by Kludex** [comment, marked as answer]
This will work for you:
```bash
curl -X POST "http://127.0.0.1:8000/" -H  "accept: application/json" -H  "Content-Type: application/json" -d "{\"link\":\"BHDH\",\"target\":\"https://google.fr\",\"extras\":\"{\\\"code\\\": \\\"ABCDE\\\"}\"}"
```

Code to test:
```python
from fastapi import FastAPI
from typing import Optional, Any, Dict
from pydantic import BaseModel, HttpUrl, Json

class Link(BaseModel):
    link: str
    target: HttpUrl
    extras: Json[Dict[str, Any]]

app = FastAPI()

@app.post("/")
async def home(link: Link):
    return link
```
Swagger input:
```json
{
  "link": "BHDH",
  "target": "https://google.fr",
  "extras": "{\"code\": \"ABCDE\"}"
}
```

But... I do believe that you don't want to do this :sweat_smile: 
That whole body is a JSON, you probably want to use `typing.Dict` instead of `pydantic.Json`.

**Comment by goldyfruit** [comment]
@Kludex ahah indeed, I prefer to avoid the `\` !!
`typing.Dict` did the trick, thanks.

**Comment by tiangolo** [comment]
Thanks for the help here @Kludex ! 👏  🙇 

Thanks for reporting back and closing the issue @goldyfruit  👍

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
